application/src/main/java/org/thingsboard/mqtt/broker/actors/device/PersistedDeviceActorMessageProcessor.java

```diff
@@ -122,7 +113,9 @@ public void processingSharedSubscriptions(SharedSubscriptionEventMsg msg) {
         if (CollectionUtils.isEmpty(msg.getSubscriptions())) {
             return;
         }
-        PacketIdAndSerialNumber lastPacketIdAndSerialNumber = getLastPacketIdAndSerialNumber();
+
+        int lastPacketId = deviceMsgService.getLastPacketId(clientId);
+        int prevPacketId = lastPacketId;
```

Comment:
sounds logical. I would still suggest removing this to have a better readability of the code since saving into redis will be a relatively cheap operation